### PR TITLE
improvement(modal): add option to prevent or allow the default event

### PR DESCRIPTION
**Pull request by brunnogomes**
Add option to prevent or allow the default event when using the modal.

The default is set to true, which is the current behavior so this doesn't break any current usage.

### CODE REVIEW

File: src/modal.js [hunk 1]
```diff
@@ -68,7 +77,9 @@ export default class extends Controller {
   }
 
   close(e) {
-    if (e) e.preventDefault();
+    if (this.preventDefaultActionClosing) {
+      if (e) e.preventDefault();
```
Comment (excid3) [inline src/modal.js:81]: Can we merge these two `if`s into one? `if (e && this.preventDefaultActionClosing) {`

Comment (brunnogomes) [reply src/modal.js]: @excid3 Done. ✔️

**Comment by excid3** [comment]
Thanks @brunnogomes 🙏

**Comment by brunnogomes** [comment]
@excid3 BTW thanks a lot for this project! It is really nice for people starting with TailwindCSS to have some components like these ready to go.

**Comment by excid3** [comment]
You're welcome, thats fantastic to hear!

**Pull request merged**